### PR TITLE
DRAFT 😵‍💫 Use Qt 6.6.3 for macos functional tests and windows/mac unit tests 

### PR DESCRIPTION
**Pull request by lesleyjanenorton**
## Description

Here we update the following jobs to use Qt 6.6:
- macos functional tests
- windows unit tests
- macos unit tests 

To follow later: 
- Linux unit and functional tests (Once we've moved / are closer to moving to static Linux builds)
- Wasm tests

## Reference

VPN-6368

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

File: tests/unit_tests/testlocalizer.cpp [hunk 1]
```diff
@@ -295,7 +295,7 @@ void TestLocalizer::formattedDate_data() {
 
   QTest::addRow("en - one hour ago")
       << "en" << QDateTime(QDate(2000, 1, 1), QTime(10, 0), QTimeZone(0))
-      << QDateTime(QDate(2000, 1, 1), QTime(9, 0), QTimeZone(0)) << "9:00 AM"
```
Comment (lesleyjanenorton) [inline tests/unit_tests/testlocalizer.cpp:298]: It strangely seems that 6.6 wants a _narrow_ no-break space here but changing this to a _narrow_ no-break white space (U+202F) causes this test to fail on Linux where we are still on 6.2.4. If anyone has immediate thoughts or guesses about why this is different in 6.6, I'd love to hear them.

**Review by jcristau — APPROVED** [review]
taskcluster bits lgtm

**Comment by oskirby** [comment]
As a part of this work, I think we also need to bump the default value of `CMAKE_OSX_DEPLOYMENT_TARGET` in the top level CMakeLists.txt file in order to keep up with the [Qt Macos](https://doc.qt.io/qt-6.5/macos.html) supported platforms.

**Comment by lesleyjanenorton** [comment]
Closing in favor of #9620 .

**Pull request closed without merging**